salt/modules/win_event.py

```diff
@@ -0,0 +1,754 @@
+
+    .. warning::
+        This function will destroy the data object and objects that data links
+        to.
+
+    Args:
+
+        data (object): The string object to encode
+
+        encoding(str): The encoding type
+
+        encode_keys(bool): If false key strings will not be turned into bytes
+
+    Returns:
+        (object): An object with the new encoding
+    """
+
+    if isinstance(data, dict):
+        new_dict = {}
+        # recursively check every item in the dict
+        for key in data:
+            item = _to_bytes(data[key], encoding)
+            if encode_keys:
+                # keys that are strings most be made into bytes
+                key = _to_bytes(key, encoding)
+            new_dict[key] = item
+        data = new_dict
+    elif isinstance(data, list):
+        new_list = []
+        # recursively check every item in the list
+        for item in data:
+            new_list.append(_to_bytes(item, encoding))
+        data = new_list
+    elif isinstance(data, tuple):
+        new_list = []
+        # recursively check every item in the tuple
+        for item in data:
+            new_list.append(_to_bytes(item, encoding))
+        data = tuple(new_list)
+    elif isinstance(data, str):
+        # encode string data to bytes
+        data = data.encode(encoding)
+
+    return data
+
+
+def _raw_time(time):
+    """
+    Will make a pywintypes.datetime into a TimeTuple.
+
+    Args:
+
+        time (ob): A datetime object
+
+    Returns:
+        TimeTuple: A TimeTuple
+    """
+
+    return TimeTuple._make(
+        (time.year, time.month, time.day, time.hour, time.minute, time.second)
+    )
+
+
+def _make_event_dict(event):
+    """
+    Will make a PyEventLogRecord into a dictionary
+
+    Args:
+
+        event (PyEventLogRecord): An event to convert to a dictionary
+
+    Returns:
+        dict: A dictionary containing the event information
+    """
+
+    event_dict = {}
+    for event_part in EVENT_PARTS:
+        # get object value and add it to the event dict
+        event_dict[event_part] = getattr(
+            event, event_part[0].upper() + event_part[1:], None
+        )
+
+    # format items
+    event_dict["eventID"] = winerror.HRESULT_CODE(event_dict["eventID"])
+    if event_dict["sid"] is not None:
+        event_dict["sid"] = event_dict["sid"].GetSidIdentifierAuthority()
+    event_dict["timeGenerated"] = _raw_time(event_dict["timeGenerated"])
+    event_dict["timeWritten"] = _raw_time(event_dict["timeWritten"])
+
+    return _to_bytes(event_dict)
+
+
+def _get_handle(log_name):
+    """
+    Will try to open a PyHANDLE to the Event System
+
+    Args:
+
+        log_name (str): The name of the log to open
+
+    Returns:
+        PyHANDLE: A handle to the event log
+    """
+
+    # TODO: upgrade windows token
+    # "log close" can fail if this is not done
+    try:
+        return win32evtlog.OpenEventLog(None, log_name)
+    except pywintypes.error:
+        raise FileNotFoundError(
```

Comment:
The actual exception message/data is not useful?

Reply:
Now using exe.stderror